eventsource-client/src/event_parser.rs

```diff
@@ -104,7 +91,7 @@ fn logify(bytes: &[u8]) -> &str {
     }
 }
 
-fn parse_field(line: &[u8]) -> Result<Option<(&str, &[u8])>> {
+fn parse_field(line: &[u8]) -> Result<Option<(&str, &str)>> {
     if line.is_empty() {
```

Comment:
The incoming line is a series of bytes, and the result is now a tuple of UTF8 (key, value). 

It shouldn't be possible to have only a UTF8 key, but not value, since the SSE protocol requires the entire stream to be UTF8.